### DISCUSSION TITLE
Only require comic name to download

### DISCUSSION THREAD

**Discussion by vengeance101** [magnuschase293/comic_getter#25] [Ideas]
Since we're only using RCO right now, it makes sense to just pass the comic name instead of the full URL, because the URL follows a consistent pattern. 

For example, instead of passing in `https://readcomiconline.to/Comic/Joker-Last-Laugh`, we could just pass in `Joker-Last-Laugh` and have the program piece together the URL in the backend. 

One issue with this is how the single issues are loaded. We'd have to find a different way to handle:

`python3 path/to/__main__.py --single https://readcomiconline.to/Comic/Joker-Last-Laugh/Issue-4`

I recommend doing something like below, and adding a issue tag (need a better letter for this):

`python3 path/to/__main__.py --single Joker-Last-Laugh -issue 4`

This fits the API pattern established by the -s flag. which would change to:

`python3 path/to/__main__.py -i Joker-Last-Laugh -s 3`

Thoughts?

**Comment by vengeance101** [comment]
Additionally, this is going to make bulk downloads a lot easier, as we could just pass in a list of comic names instead of full URLs.

**Comment by magnuschase293** [comment]
It's a great idea. The only inconvenient to have in mind is that users will need the exact name of the comic issue/collection that is used in readcomiconline.to. Still I believe I can give support to both methods using the python's RE module to separate links from names.

**Reply by StrangeDOS** [reply]
I like it as long as it continues to support BOTH methods.  I'd have to redo a bunch of stuff in the scripts on my end to handle just working with the names.  My scripts were created to handle the URLs.

**Comment by magnuschase293** [comment]
> It's a great idea. The only inconvenient to have in mind is that users will need the exact name of the comic issue/collection that is used in readcomiconline.to. Still I believe I can give support to both methods using the python's RE module to separate links from names.

And in order to get the exact name users will still need to google for the issue/collection link and extract the name.

**Comment by vengeance101** [comment]
Xonshiz's program had a search function that was incredibly useful. Easy to see a user searching for X-Men and then requesting a list of `[Uncanny-X-Men-1963, X-Men-Inferno, X-Men-Lifedeath]`. Or single volumes they care about.

To do this though, he had access to a JSON with all the titles. There was also a way to update the JSON with new additions. I'm not sure where that JSON came from.

**Reply by StrangeDOS** [reply]
I'd love to see your tool do that.  I used to use Xonshiz's tool before he stopped updating it.

**Comment by vengeance101** [comment]
another comment here: 

current api for single issue query is:

`python3 path/to/__main__.py --single https://readcomiconline.to/Comic/Uncanny-X-Men-1963/Issue-94`

changing it to `python3 path/to/__main__.py --single Uncanny-X-Men-1963 --issue 94` might not work as intended, as the 94th item in the list is not always (or usually) issue 94.

**Comment by magnuschase293** [comment]
> Xonshiz's program had a search function that was incredibly useful. Easy to see a user searching for X-Men and then requesting a list of `[Uncanny-X-Men-1963, X-Men-Inferno, X-Men-Lifedeath]`. Or single volumes they care about.
> 
> To do this though, he had access to a JSON with all the titles. There was also a way to update the JSON with new additions. I'm not sure where that JSON came from.

I might be able to do this with selenium and using RCO built-in search engine, but I 'm not sure if it is worth the trouble. It will be just as easy to enter through the internet and get the link.

**Comment by vengeance101** [comment]
No, if that's what is required I don't think it's worth it.

**Comment by magnuschase293** [comment]
I think i found another way... scrape the comic list on RCO using selenium and then creating a .json that stores all the comic names. This could be run the first time the command is used and then it would need to be updated manually whenever the user considers it necessary. Some search algorithms should be implemented in order for the user to navigate the.json that shouldn´t be a problem.

This would allow for partial name searches (of course the more accurate the search less matches will pop up). I think by just hovering over each title I can get publication date, a short summary and jpg of the cover. I'll start working on it as soon as I can and see what's useful and what isn't.
